doc/design/networking/Investigation-on-NSX-integration.md

```diff
@@ -0,0 +1,109 @@
+# Investigation on NSX micro-segmentation via Docker network create
+This document is for this issue: https://github.com/vmware/vic/issues/3936
+
+
+## A Working Installation of NSX-v
+I documented how to install and configure NSX with Vsphere: How to set up VM-to-VM communication on NSX logical switch with Nimbus
+https://confluence.eng.vmware.com/pages/viewpage.action?pageId=209964551
+
+## An Overview of Workflow
+
+### Prerequisite
+VSphere admin already has NSX installed and configured on VCenter. That means there is an NSX logical switch created by the admin, which automatically creates a port group on distributed switch.
+So this port group can be used as a bridge network for `vic-machine create`
+
+![Distributed Switch with logic switch added] (pics/dswitch.png)
+
+### Create VCH 
+specifying 'bridge-network' as the port group 'vxw-dvs-55-virtualwire-1-sid-5000-logical-switch-1':
+  * I use this VCH to create containers, they are automatically added to this port group (i.e. logical switch)
+  * containerVMs get their IPs in the internal network
+  * containerVMs can ping each other
+
+### docker network create network-a
+Our goal:
+  * VMs can communicate with VMs in the same security group.
+  * VMs cannot talk to VMs in a differrent security group.
+  * one VM to be added to more than one security group.
```

Comment:
Yes. We can associate a NIC with a security group by configuring dynamic membership as `entity belong-to <the vNIC>`.
I just experimented with VMs with multiple vNICs. The policy on one vNIC does not affect another vNIC.
If I understand correctly, you mean a containerVM may be attached to some external container-network and also have an identity in VIC created internal network?  I will need to verify if it works in this case.